runner/sidecar/source/source.go

```diff
@@ -3,13 +3,14 @@ package source
 import (
 	"context"
 	"io"
+	"time"
 )
 
 type Interface interface {
 	io.Closer
 }
 
-type Process func(ctx context.Context, msg []byte) error
+type Process func(ctx context.Context, msg []byte, ts time.Time) error
```

Comment:
I don't think you need to change this signature. You can extract time from context using `MetaFromContext`.